fintech-examples/fintech-ui/src/app/services/auth.service.ts

```diff
@@ -15,46 +15,48 @@ export class AuthService {
   constructor(
     private router: Router,
     private finTechAuthorizationService: FinTechAuthorizationService,
-    private cookieService: DocumentCookieService
+    private cookieService: DocumentCookieService,
+    private storageService: StorageService
   ) {}
 
   login(credentials: Credentials): Observable<boolean> {
-    LocalStorage.logout();
     return this.finTechAuthorizationService.loginPOST('', credentials, 'response').pipe(
       map(response => {
-        this.cookieService.getAll().forEach(cookie => console.log('cookie after login :' + cookie));
-        LocalStorage.login(response.headers.get(Consts.HEADER_FIELD_X_XSRF_TOKEN));
-        if (!LocalStorage.isLoggedIn()) {
-          console.log('login not sucessfull');
-          this.openLoginPage();
-        }
-        localStorage.setItem(Consts.LOCAL_STORAGE_USERNAME, credentials.username);
+        this.setSessionData(response, credentials);
         return response.ok;
       })
     );
   }
 
   logout(): Observable<any> {
```

Comment:
Maybe we cam have logout function as a Promise instead of Observable, since we do not response from log out

Reply:
Yes it will a better approach